spec/ddtrace_integration_spec.rb

```diff
@@ -0,0 +1,91 @@
+require 'spec_helper'
+
+require 'ddtrace/opentracer'
+require 'datadog/statsd'
+
+RSpec.describe 'ddtrace integration' do
+  context 'after shutdown' do
+    subject(:shutdown!) { Datadog.shutdown! }
+
+    before do
+      Datadog.configure do |c|
+        c.diagnostics.health_metrics.enabled = true
+      end
+
+      shutdown!
+    end
+
+    after do
+      Datadog.configuration.diagnostics.health_metrics.reset!
+    end
+
+    context 'calling public apis' do
+      it 'does not error on tracing' do
+        span = Datadog.tracer.trace('test')
+
+        expect(span.finish).to be_truthy
+      end
+
+      it 'does not error on tracing with block' do
+        value = Datadog.tracer.trace('test') do |span|
+          expect(span).to be_a(Datadog::Span)
+          :return
+        end
+
+        expect(value).to be(:return)
+      end
+
+      it 'does not error on logging' do
+        expect(Datadog.logger.info('test')).to be_truthy
```

Comment:
Minor: This can be `be true`, see <https://github.com/ruby/ruby/blob/v3_0_0/lib/logger.rb#L477>.